### DISCUSSION TITLE
Is figure numbering working as intended with auto placement?

### DISCUSSION THREAD

**Discussion by JuliDi** [typst/typst#3549] [Q&A] [answered]
Hi,

I am not sure whether this really is a bug or working as designed, so I am asking here before opening an issue.

Consider the following document with a large (Figure 1) and a small (Figure 2) figure and some text, and figure placement set to `auto`:
```typst
#set figure(numbering: "1.1")
#set figure(placement: auto)
#lorem(100)
#figure(rect(height: 20cm, width: 10cm), caption: "Figure 1")<fig1>
#lorem(10)
#figure(rect(height: 5cm, width: 10cm), caption: "Figure 2")<fig2>
```
Figure 2 is then placed before Figure 1 (which I think is okay because it makes sense to use the space more efficiently) but also _named_ "Figure 2" – where I would expect it to get the name "Figure 1" because it's the first to occur in the document.

![Screenshot 2024-03-04 at 17 57 38](https://github.com/typst/typst/assets/20155974/993bd623-9b32-4920-af94-284d37e905dc)

Is this intentional? And if so, is there a way to get the figures to be numbered strictly in ascending order of occurence?

**Comment by laurmaedje** [comment, marked as answer]
It's a bug. It's already fixed, but the fix is not yet released.
